### PR TITLE
Updated to support Trafikverket API v2

### PR DESCRIPTION
**Pull request by emiltorp**
Older versions of the API from Trafikverket will shut down this summer:
https://api.trafikinfo.trafikverket.se/DynamicContent/ContentDetails/61c33a304f02c133ec325ef2

So here is a PR with some updates to get on the latest version

* Timestamps in v2 have more precision, so I split that to keep the same pytrafikverket interface as before
* Added support for schemaversion, and added the latest schemaversion to all apis used
* Updated the readme to include the async-change made previously

I didn't know your versioning strategy so I leave that to the maintainers to update

### CODE REVIEW

**Comment by gjohansson-ST** [comment]
Have you teated this with Home Assistant?

**Comment by emiltorp** [comment]
No, I have not. I am not a Home Assistant user at the moment. I only tested the library directly. Do any of the changes conflict with the HA integrations?

**Comment by gjohansson-ST** [comment]
Ok. Then I will test. Just wanted to check. Thanks

File: pytrafikverket/trafikverket.py [hunk 1]
```diff
@@ -106,8 +106,8 @@ def generate_node(self, parent_node):
 class Trafikverket(object):
     """Class used to communicate with trafikverket api."""
 
-    _api_url = "http://api.trafikinfo.trafikverket.se/v1.2/data.xml"
-    date_time_format = "%Y-%m-%dT%H:%M:%S"
+    _api_url = "https://api.trafikinfo.trafikverket.se/v2/data.xml"
+    date_time_format = "%Y-%m-%dT%H:%M:%S.%f%z"
```
Comment (gjohansson-ST) [inline pytrafikverket/trafikverket.py:110]: It says on the API website that only modified time has the additional "z" as it's UTC while all other times are Europe/Stockholm so why are we changing this?

File: pytrafikverket/__main__.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 GET_FERRY_ROUTE = "get-ferry-route"
 SEARCH_FOR_FERRY_ROUTE = "search-for-ferry-route"
 GET_NEXT_FERRY_STOP = "get-next-ferry-stop"
-
+DATE_TIME_INPUT = "%Y-%m-%dT%H:%M:%S"
```
Comment (gjohansson-ST) [inline pytrafikverket/__main__.py:22]: Same question here. Why different than in the other file?

**Review by gjohansson-ST — CHANGES_REQUESTED** [review]
See comments

File: pytrafikverket/__main__.py (hunk 2 above)
Comment (emiltorp) [reply pytrafikverket/__main__.py:22]: As you noticed, there was a change at the API to include timezone and milliseconds, but I wanted to keep the pytrafikverket syntax intact to (try) not to make any breaking changes, therefore I needed to have a difference between how the dates was interpreted.

Comment (gjohansson-ST) [reply pytrafikverket/__main__.py:22]: That's fine. Quite annoying that the website says something and when you actually test you get a different result.
Will try to test this within the next days to ensure it's working before it's approved and merged.

**Review by gjohansson-ST — APPROVED** [review]

**Pull request merged**